Small fixes

yougotwill opened the pull request:
Some small fixes I made on a personal branch a while back. Since the `dev` branch now compiles on my machine I thought I would make a pr.

1. On macOS, the titlebar is now in the traditional style. It will remain the same on other OSs.

- Normal OS titlebar
<img width="1440" alt="regular" src="https://user-images.githubusercontent.com/14887287/103450988-94b30b80-4d01-11eb-82f5-db20b21f141a.png">

- macOS titlebar
<img width="1440" alt="macos" src="https://user-images.githubusercontent.com/14887287/103450992-9d0b4680-4d01-11eb-808b-52b7f5070898.png">

2. Updated the menubar options on macOS to allow for hiding and using services. On other OSs it will remain the same.
<img width="1440" alt="contextmenu" src="https://user-images.githubusercontent.com/14887287/103450998-bd3b0580-4d01-11eb-8964-38121302011e.png">

3. The sidebar toggle shortcut wasn't working. It's now fixed.
<video width="1440" alt="menutoggle" src="https://user-images.githubusercontent.com/14887287/103451170-03916400-4d04-11eb-9a9f-41c2e82c3b35.mp4">

Sn8z commented:
Hey @yougotwill!
Thanks for the PR :man_dancing: 

I like the possibility to change "direction" for the titlebar, it's a good feature and I think it should be included with an option to choose if you want it or not. The option part I can fix :smile: 

Regarding the sidebar toggle, is this something that Mac software usually does? What is the purpose of being able to hide the sidebar on large screens? Is the "option + E" shortcut the most used one for this kind of feature? I'm not a daily or experienced Mac user so if there's any source where I can read about this it would be great :+1: 

Will check the PR more closely in the coming days but I like the look of it!

yougotwill commented:
> Hey @yougotwill!
> Thanks for the PR 🕺
> 
> I like the possibility to change "direction" for the titlebar, it's a good feature and I think it should be included with an option to choose if you want it or not. The option part I can fix 😄

Glad you like it. I'll try and make the toggle this weekend.

> Regarding the sidebar toggle, is this something that Mac software usually does? What is the purpose of being able to hide the sidebar on large screens? Is the "option + E" shortcut the most used one for this kind of feature? I'm not a daily or experienced Mac user so if there's any source where I can read about this it would be great 👍

To be honest. I was just checking out the shortcuts available and noticed the sidebar shortcut. When it didn't work I fixed it. I don't really have an opinion. I don't think `option+e` is a common mapping but it was in the code when I fixed the bug. I know VS Code is `Ctrl/Cmd + B` but more research probably needs to be done to decide on a good shortcut for it.

> Will check the PR more closely in the coming days but I like the look of it!
Thanks. Let me add that toggle this weekend before you merge!
